src/darwin/Framework/CHIP/MTROTAImageTransferHandler.mm

```diff
@@ -284,6 +292,18 @@ - (void)SetOtaImageTransferHandler:(MTROTAImageTransferHandler *)otaImageTransfe
         return;
     }
 
+    auto * device = [MTRDevice deviceWithNodeID:nodeId controller:controller];
+
+    MATTER_LOG_METRIC(kMetricOTADeviceVendorID, [device vendorID].unsignedIntValue);
+    MATTER_LOG_METRIC(kMetricOTADeviceProductID, [device productID].unsignedIntValue);
+    MATTER_LOG_METRIC(kMetricOTADeviceUsesThread, mIsPeerNodeAKnownThreadDevice);
+    MATTER_LOG_METRIC(kMetricOTATNumBytesProcessed, uint32_t(mNumBytesProcessed));
+    MATTER_LOG_METRIC_END(kMetricOTATransfer, error);
+
+    // Always collect the metrics to avoid unbounded growth of the stats in the collector
+    MTRMetrics * metrics = [[MTRMetricsCollector sharedInstance] metricSnapshotForCategory:@("ota") removeMetrics:TRUE];
```

Comment:
```suggestion
    // Always collect the metrics to avoid unbounded growth of the stats in the collector
    MTRMetrics * metrics = [[MTRMetricsCollector sharedInstance] metricSnapshotForCategory:@("ota") removeMetrics:YES];
```